samples/SampleEncode.py

```diff
@@ -114,28 +125,77 @@ def encode(gpuID, decFilePath, encFilePath, width, height):
     print(framesFlushed, " frame(s) received during encoder flush.")
 
 
+import argparse
+import os
+import pathlib
+
+
 if __name__ == "__main__":
 
-    print(
-        "This sample encodes first ",
-        total_num_frames,
-        " frames of input raw NV12 file to H.264 video on given GPU.",
+
+
+    parser = argparse.ArgumentParser(
+        "This sample encodes first " + str(total_num_frames) + "frames of input raw NV12 file to H.264 video on given "
+                                                               "GPU." + "\n "
+        + "Requires the GL Utility Toolkit (GLUT) and pyCUDA compiled with GL support\n"
+        + "It reconfigures encoder on-the fly to illustrate bitrate change,"
+        + " IDR frame force and encoder reset.\n"
     )
-    print(
-        "It reconfigures encoder on-the fly to illustrate bitrate change, IDR frame force and encoder reset."
+    parser.add_argument(
+        "gpu_id",
```

Comment:
in general, CLI arguments are typically using Lisp case (e.g. `gpu-id`). argparse will auto-convert `gpu-id` to `gpu_id`